Monika After Story/game/zz_submods.rpy

```diff
@@ -0,0 +1,299 @@
+                    NOTE: capital letters will be forced to lower and spaces will be replaced with underscores
+                    NOTE: Update labels MUST accept a version parameter, defaulted to the version of the label
+                    For example:
+                        author name: MonikaAfterStory,
+                        submod name: Example Submod
+                        submod vers: 1.2.3
+
+                    becomes:
+                        label monikaafterstory_example_submod_v1_2_3(version="v1_2_3")
+            """
+            #First make sure this name us unique
+            if name in submod_map:
+                store.mas_utils.writelog("[SUBMOD ERROR]: A submod with name '{0}' already exists. Ignoring.\n".format(name))
+                return
+
+            #Now we verify that the version number is something proper
+            try:
+                map(int, version.split('.'))
+            except:
+                raise SubmodError("Version number '{0}' is invalid.".format(version))
+
+            #With verification done, let's make the object
+            self.author = author
+            self.name = name
+            self.description = description if description is not None else ""
+            self.version = version
+            self.dependencies = dependencies
+            self.settings_pane = settings_pane
+            self.version_updates = version_updates
+
+            #Now we add these to our maps
+            submod_map[name] = self
+
+            #NOTE: We check for things having updated later so all update scripts get called together
+            if name not in persistent._mas_submod_version_data:
+                persistent._mas_submod_version_data[name] = version
+
+        def getVersionNumberList(self):
+            """
+            Gets the version number as a list of integers
+
+            OUT:
+                List of integers representing the version number
+            """
+            return map(int, self.version.split('.'))
+
+        def hasUpdated(self):
+            """
+            Checks if this submod instance was updated (version number has incremented)
+
+            OUT:
+                boolean:
+                    - True if the version number has incremented from the persistent one
+                    - False otherwise
+            """
+            old_vers = persistent._mas_submod_version_data.get(self.name)
+
+            #If we don't have an old vers, we're installing for the first time and aren't updating at all
+            if not old_vers:
+                return False
+
+            try:
+                old_vers = map(int, old_vers.split('.'))
+
+            #Persist data was bad, we'll replace it with something safe and return False as we need not check more
+            except:
+                persistent._mas_submod_version_data[self.name] = Submod.FB_VERS_STR
+                return False
+
+            return self.checkVersions(old_vers) == 1
```

Comment:
```suggestion
            return self.checkVersions(old_vers) > 0
```
-1, 0, 1 might be what it returns, but the norm is that compareTo-style functions should not be assumed to return -1 or 1. It's always less than or greater than 0.